packages/platforms/accton/x86-64/as7946-30xb/onlp/builds/x86_64_accton_as7946_30xb/module/src/fani.c

```diff
@@ -167,6 +167,17 @@ _onlp_fani_info_get_fan_on_psu(int pid, onlp_fan_info_t* info)
 
     info->status |= ONLP_FAN_STATUS_PRESENT;
 
+    /* Get power good status */
+    ret = onlp_file_read_int(&val, "%s""psu%d_power_good", PSU_SYSFS_PATH, pid);
```

Comment:
Do not read PSU's  sysfs directly in the FAN.c, please add in the platform_lib.c.
You can refer to other projects.